### PR TITLE
Implemented the `skuba addon upgrade apply` command

### PR DESCRIPTION
**Pull request by mssola**
## Why is this PR needed?

Applies an upgrade for outdated addons. It also adds the `terraform.d` directory in the .gitignore for `ci/infra/libvirt`, which might be used in some cases when creating the infrastructure with libvirt.

## What does this PR do?

Add a new command `skuba addon upgrade apply`.

## Anything else a reviewer needs to know?

Can be tested by manipulating [these numbers](https://github.com/SUSE/skuba/blob/master/internal/pkg/skuba/kubernetes/versions.go#L92-L96)

## Tests

To be done in the form of end-to-end tests.

## Docs

Added a new man page and linked it on the main man page.

### CODE REVIEW

File: ci/infra/libvirt/.gitignore [hunk 1]
```diff
@@ -1 +1,2 @@
 *.qcow2
+terraform.d/
```
Comment (ereslibre) [inline ci/infra/libvirt/.gitignore]: Can we add this one one level up? (in `ci/infra/.gitignore`)

File: pkg/skuba/actions/addon/upgrade/apply.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	"github.com/SUSE/skuba/internal/pkg/skuba/kubeadm"
+	"github.com/SUSE/skuba/internal/pkg/skuba/kubernetes"
+	"github.com/SUSE/skuba/internal/pkg/skuba/upgrade/addon"
+	"github.com/SUSE/skuba/pkg/skuba"
+	"github.com/pkg/errors"
+)
+
+// Apply implements the `skuba addon upgrade apply` command.
+func Apply() error {
+	fmt.Printf("%s\n", skuba.CurrentVersion().String())
```
Comment (ereslibre) [inline pkg/skuba/actions/addon/upgrade/apply.go]: Is this intended or a leftover? :)

Comment (mssola) [reply pkg/skuba/actions/addon/upgrade/apply.go]: Intended: I've seen it on `addon upgrade plan`, in `node upgrade apply` and so on, so I just assumed that this was a common thing to do. If this is not the case I can remove this line :+1:

Comment (MaximilianMeister) [reply pkg/skuba/actions/addon/upgrade/apply.go]: oh i think it's a leftover and we forgot to remove it, i guess it needs to be dropped also on the other functions

Comment (MaximilianMeister) [reply pkg/skuba/actions/addon/upgrade/apply.go]: I've added a commit to drop all of them in https://github.com/SUSE/skuba/pull/733 - so I guess you can remove this one here too

Comment (mssola) [reply pkg/skuba/actions/addon/upgrade/apply.go]: Done

File: pkg/skuba/actions/addon/upgrade/apply.go [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+			ClusterVersion: currentClusterVersion,
+			ControlPlane:   clusterConfiguration.ControlPlaneEndpoint,
+			ClusterName:    clusterConfiguration.ClusterName,
+		}
+		if err := addons.DeployAddons(addonConfiguration, addons.AlwaysRender); err != nil {
+			return errors.Wrap(err, "[apply] Failed to deploy addons")
+		}
+		fmt.Printf("[apply] Successfully upgraded addons\n")
+	} else {
+		fmt.Println("Congratulations! Addons are already at the latest version available")
```
Comment (MaximilianMeister) [inline pkg/skuba/actions/addon/upgrade/apply.go]: can we make this more explicit?

```suggestion
		fmt.Printf("Congratulations! Addons for %s are already at the latest version available\n", currentVersion)
```

I've done the same for the `plan` where this explicitness is really helpful https://github.com/SUSE/skuba/pull/733

Comment (mssola) [reply pkg/skuba/actions/addon/upgrade/apply.go]: Sure :+1:

File: pkg/skuba/actions/addon/upgrade/apply.go [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+	if addon.HasAddonUpdate(updatedAddons) {
+		addonConfiguration := addons.AddonConfiguration{
+			ClusterVersion: currentClusterVersion,
+			ControlPlane:   clusterConfiguration.ControlPlaneEndpoint,
+			ClusterName:    clusterConfiguration.ClusterName,
+		}
+		if err := addons.DeployAddons(addonConfiguration, addons.AlwaysRender); err != nil {
+			return errors.Wrap(err, "[apply] Failed to deploy addons")
+		}
+		fmt.Printf("[apply] Successfully upgraded addons\n")
```
Comment (MaximilianMeister) [inline pkg/skuba/actions/addon/upgrade/apply.go]: MINOR: could use `Println` here too

Comment (mssola) [reply pkg/skuba/actions/addon/upgrade/apply.go]: Done!

**Pull request merged**